internal/git-remote-gittuf/main.go

```diff
@@ -0,0 +1,248 @@
+		  the remote has and what their tips point to
+		  Note: we interpose this to learn the status of gittuf refs on the
+		  remote
+		* Git negotiates with git-upload-pack the objects it wants based on the
+		  refs that must be fetched
+		  Note: we interpose this to request gittuf specific objects as well
+		* The remote sends a packfile with the requested objects
+		* git-remote-gittuf uses update-ref to set the local gittuf refs, as Git
+		  will not do this for us
+
+	Anatomy of a push:
+		* Git invokes list for-push (protocol v0/v1) to list the refs available
+		  on the remote
+		* Git invokes push (protocol v0/v1) to indicate what refs must be
+		  updated to on the remote
+		* A packfile is created and streamed to git-receive-pack on the server
+*/
+
+var (
+	// logFile is used to debug git-remote-gittuf. It is set using
+	// GITTUF_LOG_FILE.
+	logFile io.Writer
+
+	// gitVersion contains the version of Git used by the client invoking
+	// git-remote-gittuf. It is used to self-identify with a remote service such
+	// as git-upload-pack and git-receive-pack.
+	gitVersion string
+
+	isPacketMode = false
+
+	flushPkt     = []byte{'0', '0', '0', '0'}
+	delimiterPkt = []byte{'0', '0', '0', '1'}
+	endOfReadPkt = []byte{'0', '0', '0', '2'}
+
+	zeroHash = "0000000000000000000000000000000000000000"
+)
+
+type state int
+
+// TODO: we have a mix of states here, lsRefs onwards are when interacting with
+// upload-pack.
+const (
+	start                   state = iota // top level menu of helper, serves as router too when not within a service
+	serviceRouter                        // router for service
+	lsRefs                               // upload-pack
+	lsRefsResponse                       // upload-pack
+	requestingWants                      // upload-pack
+	requestingWantsResponse              // upload-pack
+	packfileIncoming                     // upload-pack
+	packfileDone
+)
+
+const (
+	gitUploadPack   = "git-upload-pack"
+	gitReceivePack  = "git-receive-pack"
+	gittufRefPrefix = "refs/gittuf/"
+)
+
+func run() error {
+	if len(os.Args) < 3 {
+		return fmt.Errorf("usage: %s <remote-name> <url>", os.Args[0])
+	}
+
+	gitDir := os.Getenv("GIT_DIR")
+	remoteName := os.Args[1]
+	url := os.Args[2]
+
+	var handler func(string, string) (map[string]string, bool, error)
+	switch {
+	case strings.HasPrefix(url, "https://"), strings.HasPrefix(url, "http://"), strings.HasPrefix(url, "ftp://"), strings.HasPrefix(url, "ftps://"):
+		log("Prefix indicates curl remote helper must be used")
+		handler = handleCurl
+	case strings.HasPrefix(url, "/"), strings.HasPrefix(url, "file://"):
+		log("Prefix indicates file helper must be used")
+		return nil
+	default:
+		log("Using ssh helper")
+		handler = handleSSH
+	}
+
+	// var rslTip string
+	// cmd := exec.Command("git", "--git-dir", gitDir, "rev-parse", rsl.Ref) //nolint:gosec
+	// rslTipB, err := cmd.Output()
+	// if err == nil {
+	// 	rslTip = string(bytes.TrimSpace(rslTipB))
+	// }
```

Comment:
Is this supposed to still be here, or is this somthing that should be uncommented later on?

Reply:
the latter